Development

alexandrabagi opened the pull request:
First attempt to merge development branch to master branch. In the React folder, you can find our React project that we started from Tim's tutorial, we removed the unneccessary files and modified the rest. We have the main screens and a clickable element, in the next iteration we plan to make the rest of the elements clickable and create the rest of the screens.
David's FileMaker files are in the FileMaker folder.

ayanlee commented on React/src/App.js:
```diff
@@ -43,14 +43,14 @@ class App extends Component {
   }
 
-  setCategory = category => {
+  /*setCategory = category => {
```
Please, don't commit commented out code.

ayanlee commented on React/src/App.js:
```diff
@@ -94,12 +94,12 @@ class App extends Component {
                 <HourRegistrationScreen 
                   selectedCategory={this.state.currentCategory}
-                  onCategorySelect={this.setCategory}
+                  //onCategorySelect={this.setCategory}
```
Remember to delete

ayanlee commented on React/src/screens/HourRegistration.js:
```diff
@@ -2,8 +2,8 @@ import React from 'react';
-import Clock from './../images/clock.PNG'
-import { faAlignCenter } from '@fortawesome/free-solid-svg-icons';
+import Clock from './../images/Clock.PNG'
+// import { faAlignCenter } from '@fortawesome/free-solid-svg-icons';
```
Remember to delete

ayanlee commented on React/src/screens/HourRegistration.js:
```diff
@@ -2,8 +2,8 @@ import React from 'react';
+// import { faAlignCenter } from '@fortawesome/free-solid-svg-icons';
 
 
 // export const CategoryScreen = props => {
```
Remember to delete

ayanlee commented on React/src/App.js:
```diff
@@ -6,12 +6,12 @@ import {
 } from 'react-router-dom';
 
-import { Collections } from './data/Collections';
+// import { Collections } from './data/Collections';
```
Remember to delete

alexandrabagi commented:
Dear Mircea, Ayanle,
Here is the summary of the main changes in our commits up to date:
- Implemented popup window onClick for buttons Coworker and Both, we plan to refactor this part once we managed to make all the functionalities work.
- Implemented button design change depending on whether they are selected or not. 
- Deleted commented out lines. 
- Removed unused code for image filters, that we inherited from the original file. 
- Isolated the image screen as an component tha can be reused in both MaterialGridScreen and ProductGridScreen. 
- Added dynamic way of showing different items in each material categoy by using json file.
Our React project files stayed in the React folder, as the project has a separate part which is done in FileMaker, and this structure allows us to unambiguously separate project files of the different parts from each other.
Best,
App Built for Builders team

ayanlee commented on React/src/images/Plastering/data.json:
```diff
@@ -0,0 +1,13 @@
+{
+    "images": [
+        {"id": "Plastering/p1", "name": "Type a"},
+<<<<<<< HEAD
```
You have to resolve this merge conflict

alexandrabagi replied:
The conflict is now resolved :)

ayanlee commented on React/src/screens/HourRegistration.js:
```diff
@@ -1,49 +1,95 @@
-          </div>
-        </div>
+
+            <Button id = {this.state.buttonsRow1[0].id} label={this.state.buttonsRow1[0].label} selected={this.state.buttonsRow1[0].selected} onClick={()=> this.changeButtonState(this.state.buttonsRow1[0].id, 0)}/>
```
Why did you put this in an array, if we can't iterate over this?

ayanlee commented on React/src/screens/ProductsGridScreen.js:
```diff
@@ -0,0 +1,35 @@
+    toggleImageSelect = (id) => {
+      // let imagesToUpdate = [...this.state.images];
+      // let imageToUpdate = imagesToUpdate.find(image => image.id === id);
+      // window.location = './products/'+ imageToUpdate.id;
```
As a general rule, no commented out code :)

ayanlee commented:
Please fix the merge conflict to get the PR approved :)

alexandrabagi commented:
Hi! As the merge conflict has been resolved, can we merge Development branch into Master?

mircealungu commented:
@ayanlee  -- do you agree with the merging? :)

ayanlee commented:
> @ayanlee -- do you agree with the merging? :)
Yes, it is approved :)

mircealungu commented on React/src/App.js:
```diff
@@ -20,25 +20,31 @@ class App extends Component {
-    window.location = './products/'+ imageToUpdate.id;
+    let materialsToLookUp = [...this.state.materials];
+    let selectedMaterial= materialsToLookUp.find(material => material.id === id);
+    // window.location = './products/'+ selectedMaterial.id +'/' + selectedMaterial.image ;
```
is the commented out code necessary?
